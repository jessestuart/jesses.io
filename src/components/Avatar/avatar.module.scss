@import '../../styles/js-tachyons.css';
@import '../../utils/colors.scss';

.avatar {
  border-radius: 50%;
  box-shadow: 0 0 12px 12px rgba(40, 40, 40, 0.6);
  padding: 0;
  margin: 0;
}

@media (--breakpoint-not-small) {
  img.js-avatar {
    max-width: 75%;
  }
}
