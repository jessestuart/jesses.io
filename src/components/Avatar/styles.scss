@import '../../utils/colors.scss';

.js-avatar-wrapper {
  align-items: center;
  display: flex;
  // height: 100%;
  justify-content: center;
  margin: 0 auto;
  // padding-bottom: 20px;
}

img.js-avatar {
  border-radius: 50%;
  box-shadow: 0 0 5px 5px $black;
  display: flex;
  justify-content: center;
  width: 75%;
  padding: 0;
}

// @media (max-width: map-get($container-max-widths, 'xl') - 1) {
//   img.js-avatar {
//     max-width: 50%;
//   }
// }
