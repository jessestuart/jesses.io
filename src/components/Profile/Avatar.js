/* @flow */
import Img from 'gatsby-image'
import React from 'react'

import classNames from 'classnames'

const Avatar = ({ image }: { image: * }) =>
  image ? (
    <Img
      className={classNames('br-100 justify-center mv2 pa0')}
      style={{
        boxShadow: '0 0 12px 12px rgba(40, 40, 40, 0.6)',
      }}
      fluid={image}
    />
  ) : null

export default Avatar
