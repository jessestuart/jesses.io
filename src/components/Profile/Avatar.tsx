import Img, { FluidObject } from 'gatsby-image'
import React from 'react'

interface Props {
  image: FluidObject
}

const Avatar = ({ image }: Props) =>
  !image ? null : (
    <Img
      className="br-100 justify-center mv2 pa0"
      loading="eager"
      style={{
        boxShadow: '0 0 12px 12px rgba(40, 40, 40, 0.6)',
      }}
      fluid={image}
    />
  )

// const Avatar = ({ image }: Props) => (
//   <StyledImg
//     className="br-100 shadow-3"
//     padding={0}
//     justifyContent="center"
//     my={2}
//     loading="eager"
//     fluid={image}
//   />
// )

export default Avatar
