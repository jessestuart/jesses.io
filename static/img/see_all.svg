<svg xmlns="http://www.w3.org/2000/svg" width="400" height="267" viewBox="0 0 400 267">
  <g fill="none" fill-rule="evenodd">
    <rect width="400" height="266.67" fill="#EAEAEA"/>
    <text fill="#4A4A4A" font-family="WhitneyHTF-Light, Whitney HTF" font-size="64" font-weight="300">
      <tspan x="81" y="156">see all</tspan>
    </text>
    <path fill="#F63F6C" stroke="#F63F6C" stroke-linecap="square" stroke-width="5" d="M260.5 137.04168L318.5 137.04168M318.5 137.04168L307.7 134.04168 307.7 140.04168 318.5 137.04168z"/>
  </g>
</svg>
