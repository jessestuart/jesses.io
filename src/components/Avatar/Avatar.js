import React from 'react'

import styles from './avatar.module.scss'
import avatarImage from '../../../static/img/avatar_square.jpg'

const placeholder = 'http://via.placeholder.com/800x800'
// const imagepath = placeholder
const getImagePath = () => {
  const isDev = process.env.NODE_ENV === 'development'
  return avatarImage
  // if (!isDev) { return placeholder
  // }
  // const pathPrefix = isDev ? '' : __PATH_PREFIX__
}

export default function Avatar() {
  return (
    <img
      className={styles.avatar}
      src={avatarImage}
      // src={pathPrefix + '/img/avatar_square.jpg'}
    />
  )
}

// <img className={styles.avatar} src={placeholder} />
