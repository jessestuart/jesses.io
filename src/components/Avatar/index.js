import React from 'react'

import './styles.scss'

export default function Avatar() {
  const isDev = process.env.NODE_ENV === 'development'
  const pathPrefix = isDev ? '' : __PATH_PREFIX__
  return (
    <img className="js-avatar" src={pathPrefix + '/img/avatar_square.jpg'} />
  )
}
