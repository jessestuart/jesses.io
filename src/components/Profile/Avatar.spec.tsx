import React from 'react'

import renderer from 'react-test-renderer'

import img from '../../../static/img/avatar_square.jpg'
import Avatar from './Avatar'

jest.mock('gatsby-image')

test('Render Avatar component.', () => {
  const component = renderer.create(<Avatar image={img} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
