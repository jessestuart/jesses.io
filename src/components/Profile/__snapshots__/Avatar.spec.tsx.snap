// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Avatar component. 1`] = `null`;
